pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -175,9 +175,28 @@ func ensurePublicIPAddress(opt *Options, log logr.Logger, client openstackclient
 		return nil, errors.New("floatingPool must not be empty")
 	}
 
+	if infraStatus.Networks.Router.ID == "" {
+		return nil, errors.New("router must not be empty")
+	}
+
+	routers, err := client.GetRouterByID(infraStatus.Networks.Router.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(routers) == 0 {
+		return nil, fmt.Errorf("router with ID %s was not found", infraStatus.Networks.Router.ID)
+	}
+
+	var subnetID string
+	if router := routers[0]; len(router.GatewayInfo.ExternalFixedIPs) != 0 {
```

Comment:
I'm not too deep in the topic. In which case is `ExternalFixedIPs` empty and is this considered an error?

Reply:
Practically it would mean that there is no FIP attached to the router which would be an error condition for the shoot. It would be an invalid shoot state so we can treat it as an error.

Reply:
will you make another commit to return an error in this case?

Reply:
@petersutter, yes I will update